### PR TITLE
7036 zvol_swap_004_pos test failed

### PR DESCRIPTION
**Pull request by ikozhukhov**

### CODE REVIEW

**Comment by ahrens** [comment]
@zettabot go

**Comment by ahrens** [comment]
@jwk404 can you take a look?

**Comment by ikozhukhov** [comment]
could you restart build with tests - more swap tests SKIP because setup failed. i think it is issue with env and need re-check it again

**Comment by ikozhukhov** [comment]
for example: result on my env:
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/setup (run as root) [00:04] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_001_pos (run as root) [00:04] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_002_pos (run as root) [00:42] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_003_pos (run as root) [00:00] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos (run as root) [00:37] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_005_pos (run as root) [00:05] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_006_pos (run as root) [00:00] [PASS]
Test: /opt/zfs-tests/tests/functional/zvol/zvol_swap/cleanup (run as root) [00:00] [PASS]

**Comment by ahrens** [comment]
@zettabot go

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh [hunk 1]
```diff
@@ -44,8 +45,18 @@
 
 verify_runnable "global"
 
+function cleanup
+{
+	if is_swap_inuse $swapname ; then
+		log_must $SWAP -d $swapname
```
Comment (jwk404) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: If you change this function to this:

is_swap_inuse $swapname && log_must $SWAP -d $swapname
datasetexists $vol && log_must $ZFS destroy $vol

then you don't need the last two lines of this diff, where the swap gets deleted and the volume destroyed. The cleanup function will handle it automatically.

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh [hunk 2]
```diff
@@ -61,23 +72,34 @@ for vbs in 8192 16384 32768 65536 131072; do
...
+                        ((new_volsize <= mem)) || log_fail \
+                            "3:Unexpected volsize: $new_volsize, mem:$mem, min:$min, max:$max"
+                fi
 
 		log_must $SWAP -d $swapname
 		log_must $ZFS destroy $vol
 	done
 done
 
+log_must $SWAP -d $swapname
```
Comment (jwk404) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: As mentioned above, these two lines can be deleted.

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh [hunk 3]
```diff
@@ -61,23 +72,34 @@ for vbs in 8192 16384 32768 65536 131072; do
...
-		elif ((mem >= max)); then	# volsize should be 16G
-			new_volsize=$(get_prop volsize $vol)
-			((new_volsize == max)) || log_fail \
-			    "Unexpected volsize: $new_volsize"
-		else				# volsize should be 'mem'
-			new_volsize=$(get_prop volsize $vol)
-			((new_volsize == mem)) || log_fail \
-			    "Unexpected volsize: $new_volsize"
-		fi
+                if ((mem <= min)); then         # volsize should be 2G
```
Comment (jwk404) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: This whole section should be reverted. When new_volsize matches (which should happen on the first iteration) we'll do a "break 2" and skip the rest of the tests. The original code here is correct.

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh (hunk 1 above)
Comment (ikozhukhov) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: i have added this function based on another function in zfs tests - for consistency and good reading of what we try to do

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh (hunk 3 above)
Comment (ikozhukhov) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: your proposed/original function trace with 'set -x'
https://paste.ec/paste/lt2fI+gb#hlrErHUNdf8h7G1NtkhRTuiLyD+oNR5WtsNqc6zGJm-

as you can see - we are FAILed with logic.

it is trace with 'set -x' of my variant where test is pass:
https://paste.ec/paste/HKpb9c0a#ojKms-GJmqxqfnWSTmSyuIdtuYIuxNBHlOJyIh7Vq4l

File: usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh (hunk 2 above)
Comment (ikozhukhov) [inline usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh]: i have removed lines and have committed update

**Comment by ahrens** [comment]
@zettabot go

**Comment by jwk404** [comment]
LGTM

**Comment by ahrens** [comment]
@zettabot go

**Pull request closed without merging**